src/v/cluster/BUILD

```diff
@@ -121,3 +121,32 @@ redpanda_cc_library(
         "@abseil-cpp//absl/hash",
     ],
 )
+
+redpanda_cc_library(
+    name = "topic_configuration",
+    srcs = [
+        "remote_topic_properties.cc",
+        "topic_configuration.cc",
+        "topic_properties.cc",
+    ],
+    hdrs = [
+        "remote_topic_properties.h",
+        "topic_configuration.h",
+        "topic_properties.h",
+    ],
+    include_prefix = "cluster",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//src/v/cloud_storage:remote_label",
+        "//src/v/model",
+        "//src/v/pandaproxy/schema_registry:subject_name_strategy",
```

Comment:
Eek, cluster should not depend on pandaproxy! We should split off the types to models or something. I can't wait to cleanup some of this now that it's explicit what's going on.

Reply:
> cleanup some of this now that it's explicit what's going on.

yeh it'll be very satisfying